_posts/2022-08-15-python-conda-environments-create.md

```diff
@@ -0,0 +1,377 @@
+* environments
+* or tools
+
+I'll devote another blog to setting up testing. 
+
+This blog is about how to setup a consistent and easier-to-install Python (conda) environment. Setting up consistent environments can help you create better,
+easier-to-reproduce workflows. This is particularly important if you are trying
+to help students setup their computers and create workflows in Python. But it 
+will also support your scientific workflows!
+
+## My Path From Using Matlab to Teaching R & Python
+
+When I was working at NEON building the NEON data skills program, the obvious
+choice for a programming language to learn and teach was R. I had used Matlab
+throughout my PhD program to process my data. But the ecology and biology 
+community which NEON was focused on were heavily using R. And it was 
+free and open source (FOSS) so I jumped ship from Matlab to R. I couldn't afford
+a Matlab license at the time anyway so it made sense. 
+
+I quickly learned to love R. It was intuitive. 
+And once I got over the pirate symbols like `<-`  and `%>%` I really  
+enjoyed using R and the `RStudio` interface is so intuitive! 
+
+## R Came With Community 
+
+But what I loved most was the community. There was SO much support and infrastructure. 
+I could to do things like - create a new package to process a specific type of data. 
+I could write vignettes for packaging. I could easily create reproducible 
+reports with citations (`RMarkdown`).
+
+Using R, I could run a workshop and send out setup instructions. People 
+installed the packages and came ready to learn. Just like that. Out of the 
+imaginary "box". Sure `GDAL` was sometimes problematic in those year working with
+R. But I figured it out. 
+
+*(Does anyone actually put software in boxes anymore?)*
+
+## Here Comes Python... and GDAL - Help!!
+
+My move to the Python programming language was much less graceful. I moved because 
+my program at CU Boulder was rooted in the Earth sciences. The earth science 
+community was using Python more regularly than R. I determined this by surveying 
+the community - public and private sector. 
+
+My surveys told me that industry was using `Python` a lot too. It's 
+generic design lends it to be used in may different fields and applications. 
+I knew that students could benefit professionally across many fields if they knew
+how to code in `Python`. 
+
+*By the way, I still think R is great for many things!
+I still use R too! No judgement here!*
+{: .notice }
+
+### How Does One Install GDAL Anyway?
+
+But, installing Python packages was much less intuitive. Quickly I had to figure 
+out whether to use `pip` or `conda` to manage installing package. And once I 
+settled on conda, it became what channel do I use? `conda-forge` seemed like 
+a good one. I found myself with environment
+conflicts. 
+
+oh yes, conflict.
+
+I'd install one thing and it would have a conflict with something else. Id remove that
+other thing and something else would break. 
```

Comment:
```suggestion
I'd install one tool and it would have a conflict with something else. I'd remove that
tool causing the conflict and something else would break. And so it went. :sob:
```